service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GetGpcStructuredTaskExecutor.java

```diff
@@ -80,4 +108,67 @@ public void execute(GetGpcStructuredTaskDefinition structuredTaskDefinition) {
 
         detectTranslationCompleteService.beginSendingCompleteExtract(ehrExtractStatus);
     }
+
+    private Bundle  getStructuredRecord(GetGpcStructuredTaskDefinition structuredTaskDefinition) {
+        var structuredRecord = fhirParseService.parseResource(gpcClient.getStructuredRecord(structuredTaskDefinition), Bundle.class);
+        //TODO whole block below might not be needed in GPC 1.5.1
+        getPatientId(structuredTaskDefinition).ifPresent(patientId -> {
+            ehrExtractStatusService.updateEhrExtractStatusAccessDocumentPatientId(structuredTaskDefinition, patientId);
+            // calling getDocumentReferencesBundle because currently getStructuredRecord doesn't have document references
+            var documentReferences = getDocumentReferencesBundle(structuredTaskDefinition, patientId);
+
+            ResourceExtractor.extractResourcesByType(documentReferences, DocumentReference.class)
+                .map(resource -> new Bundle.BundleEntryComponent().setResource(resource))
+                .forEach(bundleEntry -> structuredRecord.getEntry().add(bundleEntry));
+            structuredRecord.setTotal(structuredRecord.getEntry().size());
+        });
+        return structuredRecord;
+    }
+
+    private void queueGetDocumentsTask(TaskDefinition taskDefinition, List<OutboundMessage.ExternalAttachment> externalAttachments) {
+        externalAttachments.stream()
+            .map(externalAttachment -> buildGetDocumentTask(taskDefinition, externalAttachment))
+            .forEach(taskDispatcher::createTask);
+    }
+
+    private Optional<String> getPatientId(GetGpcStructuredTaskDefinition taskDefinition) {
+        var response = gpcClient.getPatientRecord(taskDefinition);
+
+        var ctx = FhirContext.forDstu3();
+        var parser = ctx.newJsonParser();
+
+        var fhirBundle = parser.parseResource(Bundle.class, response);
+
+        return ResourceExtractor.extractResourcesByType(fhirBundle, Patient.class)
+            .map(Resource::getIdElement)
+            .map(IdType::getIdPart)
+            .reduce((a, b) -> {
+                throw new IllegalStateException("There is more than 1 Patient resource in Structured Record Bundle");
+            });
+    }
+
+    private Bundle getDocumentReferencesBundle(GetGpcStructuredTaskDefinition taskDefinition, String patientId) {
+        var response = gpcClient.getDocumentReferences(taskDefinition, patientId);
+
+        var ctx = FhirContext.forDstu3();
```

Comment:
there's no need to initialize FhirContext each time - we could use FhirParseService class